include/swift/AST/AvailabilityDomain.h

```diff
@@ -236,6 +242,10 @@ class AvailabilityDomain final {
   /// Returns the string to use when printing an `@available` attribute.
   llvm::StringRef getNameForAttributePrinting() const;
 
+  /// Returns the decl that represents the domain, or `nullptr` if the domain
+  /// does not have a decl.
+  Decl *getDecl() const;
```

Comment:
For the moment, does this always points to a clang VarDecl? And when would a domain not have a decl?

Reply:
Yes, right now it can only reference and import `VarDecl`. However, I anticipate it pointing to additional types of declarations once we have a Swift syntax for custom availability domains.